examples/geometry/surfaces/examplePlaneProbingSurfaceLocalEstimator.cpp

```diff
@@ -0,0 +1,162 @@
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file
+ * @ingroup Examples
+ * @author Jocelyn Meyron (\c jocelyn.meyron@liris.cnrs.fr )
+ * Laboratoire d'InfoRmatique en Image et Systemes d'information - LIRIS (CNRS, UMR 5205), CNRS, France
+ *
+ * @date 2020/12/07
+ *
+ * An example file that shows how to construct and use a PlaneProbingDigitalSurfaceLocalEstimator.
+ *
+ * This file is part of the DGtal library.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <unordered_map>
+#include "ConfigExamples.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/base/Common.h"
+#include "DGtal/geometry/surfaces/estimation/PlaneProbingTetrahedronEstimator.h"
+#include "DGtal/geometry/surfaces/estimation/PlaneProbingParallelepipedEstimator.h"
+#include "DGtal/geometry/surfaces/DigitalSurfacePredicate.h"
+#include "DGtal/geometry/surfaces/estimation/PlaneProbingDigitalSurfaceLocalEstimator.h"
+#include "DGtal/helpers/Shortcuts.h"
+#include "DGtal/helpers/ShortcutsGeometry.h"
+#include "DGtal/io/viewers/Viewer3D.h"
+///////////////////////////////////////////////////////////////////////////////
+
+using namespace std;
+using namespace DGtal;
+
+using KSpace           = Z3i::KSpace;
+using SH3              = Shortcuts<KSpace>;
+using SHG3             = ShortcutsGeometry<KSpace>;
+using Surface          = SH3::DigitalSurface;
+using Surfel           = SH3::Surfel;
+using Point            = SH3::Point;
+using RealPoint        = SH3::RealPoint;
+using Integer          = Point::Coordinate;
+
+RealPoint centerSurfel (KSpace const& K, SH3::SCell const& s)
+{
+    auto pointels = SH3::getPrimalVertices(K, s, true);
+
+    Point p = K.uCoords(pointels[0]),
+          u = K.uCoords(pointels[1]) - p,
+          v = K.uCoords(pointels[3]) - p;
+
+    static const RealPoint shift(-0.5, -0.5, -0.5);
+
+    return p + 0.5 * u + 0.5 * v + shift;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+int main( int argc, char** argv )
+{
+    QApplication application(argc, argv);
+
+    std::string volfile = (argc > 1) ? argv[1] : (examplesPath + "samples/cat10.vol");
+
+    auto params    = SH3::defaultParameters() | SHG3::defaultParameters();
+    auto bimage    = SH3::makeBinaryImage(volfile, params);
+    auto K         = SH3::getKSpace(bimage, params);
+    auto surface   = SH3::makeDigitalSurface(bimage, K, params);
+    auto surfels   = SH3::getSurfelRange(surface);
+
+    Integer bound = params["maxAABB"].as<Integer>();
+    double gridstep = params["gridstep"].as<double>();
+
+    Viewer3D<> viewer(K);
+    viewer << SetMode3D(Surfel().className(), "Basic");
+    viewer.show();
+
+    //! [PlaneProbingDigitalSurfaceLocalEstimatorConstruction]
+    using SurfacePredicate = DigitalSurfacePredicate<Surface>;
+    using ProbingAlgorithm = PlaneProbingParallelepipedEstimator<SurfacePredicate, ProbingMode::R1>;
```

Comment:
Yes, you understood correctly: it's the internal probing algorithm that is used on each surfel. I'm not sure about "ProbingEstimator" since `PlaneProbingSurfaceLocalEstimator` is also a kind of "probing estimator". Maybe InternalProbingAlgorithm?

Reply:
Yes